src/elements/GenresList.tsx

```diff
@@ -1,41 +1,28 @@
-import React from "react";
-import { Selection } from "@react-types/shared";
-import { IGenreData } from "../models/genreData";
-import { Listbox, ListboxItem } from "@nextui-org/react";
+import { getGenres } from "../utilities/api";
+import { useQuery } from "@tanstack/react-query";
 
-function GenreList({ genres }: { genres: IGenreData[] }) {
-  //   state to track the selected genres
-  const [selectedKeys, setSelectedKeys] = React.useState<Selection>(
-    new Set([""])
-  );
+function GenreList() {
+  const { isPending, isLoading, isError, data } = useQuery({
+    queryKey: ["genres"],
+    queryFn: getGenres,
+  });
 
-  // join the selected genres to get a search query
-  const selectedValue = React.useMemo(
-    () => Array.from(selectedKeys).join(", "),
-    [selectedKeys]
-  );
+  if (isPending || isLoading) return <h1>Loading....</h1>;
+  if (isError) return <h1>Opps! Error loading data!</h1>;
 
-  return (
-    <>
-      <h1>Explore genres</h1>
-      <div className="flex flex-col gap-2">
-        <div className="w-full border-small px-1 py-2 rounded-small border-default-200 dark:border-default-100">
-          <Listbox
-            aria-label="Multiple selection example"
-            variant="flat"
-            disallowEmptySelection
-            selectionMode="multiple"
-            selectedKeys={selectedKeys}
-            onSelectionChange={setSelectedKeys}
-          >
-            {genres.map((genre) => (
-              <ListboxItem key={genre.genreName}>{genre.genreName}</ListboxItem>
-            ))}
-          </Listbox>
+  if (data) {
+    const genres = data;
```

Comment:
Instead of having `const genres = data ` you can change the code in your destructuring of the useQuery to ` const {isPending, isLoading, isError, data : genre } = useQuery(...) ` in order to rename the data property/object